### PR TITLE
gsm: upgrade version 1.0.19 -> 1.0.22

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- gsm: upgrade version 1.0.19 -> 1.0.22
- gsm: add newline EOF

###### Change Log  

- gsm: upgrade version 1.0.19 -> 1.0.22
- gsm: add newline EOF

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: [PR-7192](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=479937&view=results)

Local Build:

image

### CODE REVIEW

File: SPECS/gsm/gsm-makefile.patch [hunk 1]
```diff
@@ -15,7 +16,7 @@ diff -up gsm-1.0-pl17/Makefile.mk gsm-1.0-pl17/Makefile
  
 +SO_MAJOR = 1
 +SO_MINOR = 0
-+SO_PATCH = 18
++SO_PATCH = 19
```
Comment (reubeno) [inline SPECS/gsm/gsm-makefile.patch:19]: Should this be 22 as well? Did we hand-port this patch or did we get an updated copy of the patch from somewhere?

Comment (mfrw) [reply SPECS/gsm/gsm-makefile.patch]: Ofcourse .. thank you :)

**Review by mfrw — COMMENTED** [review]
New Test Build:
- [PR-7192](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=483926&view=results)

**Review by kanikanema — APPROVED** [review]
LGTM

**Pull request merged**